contracts/lib/groth16-verifiers/Groth16VerifierAuthV2Wrapper.sol

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2017 Christian Reitwiessner
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom
+// the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+// PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+// 2019 OKIMS
+//      ported to solidity 0.6
+//      fixed linter warnings
+//      added requiere error messages
+//
+//
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.27;
+
+import {Groth16VerifierAuthV2} from "./Groth16VerifierAuthV2.sol";
+import {IVerifier} from "../../interfaces/IVerifier.sol";
+
+contract Groth16VerifierAuthV2Wrapper is Groth16VerifierAuthV2, IVerifier {
+    /**
+     * @dev Number of public signals for atomic mtp circuit
+     */
+    uint256 constant PUBSIGNALS_LENGTH = 3;
+
+    /**
+     * @dev Verify the circuit with the groth16 proof π=([πa]1,[πb]2,[πc]1).
+     * @param a πa element of the groth16 proof.
+     * @param b πb element of the groth16 proof.
+     * @param c πc element of the groth16 proof.
+     * @param input Public inputs of the circuit.
+     * @return r true if the proof is valid.
+     */
+    function verify(
+        uint256[2] calldata a,
+        uint256[2][2] calldata b,
+        uint256[2] calldata c,
+        uint256[] calldata input
+    ) public view returns (bool r) {
+        uint[PUBSIGNALS_LENGTH] memory pubSignals;
+
+        require(input.length == PUBSIGNALS_LENGTH, "expected array length is 11");
```

Comment:
Not 11 but 3 :)
That may be better to switch to parametrised custom error to avoid such mistakes (second time in my experience) But that's to do everywhere and obviously not in this PR.

Reply:
fixed